Packet++/header/GvcpLayer.h

```diff
@@ -0,0 +1,758 @@
+#pragma once
+
+#include "IpAddress.h"
+#include "Layer.h"
+#include "MacAddress.h"
+#include <iostream>
+
+/// @file
+
+/// This GVCP implementation is based on GigE Vision ® Specification version 2.0
+
+/**
+ * @namespace pcpp
+ * @brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+	uint16_t hostToNet16(uint16_t val);  // forward declare the function in SystemUtil.h
+	uint16_t netToHost16(uint16_t val);  // forward declare the function in SystemUtil.h
+
+	namespace internal
+	{
+		static constexpr size_t kGvcpMagicNumber = 0x42;
+		static constexpr size_t kGvcpRequestHeaderLength = 8;
+		static constexpr size_t kGvcpAckHeaderLength = 8;
+		static constexpr size_t kGvcpDiscoveryBodyLength = 248;
+		static constexpr size_t kGvcpForceIpBodyLength = 56;
+	}  // namespace internal
+
+	using GvcpFlag = uint8_t;  // flag bits are specified by each command
+
+	/// @brief Gvcp command defines the command values and the corresponding acknowledge values
+	/// See more in the spec "18 Command and Acknowledge Values"
+	enum class GvcpCommand : uint16_t
+	{
+		// Discovery Protocol Control
+		DiscoveredCmd = 0x0002,
+		DiscoveredAck = 0x0003,
+		ForceIpCmd = 0x0004,
+		ForceIpAck = 0x0005,
+
+		// Streaming Protocol Control
+		PacketResendCmd = 0x0040,
+		PacketResendAck = 0x0041,  // Resent packet must be on the stream channel
+
+		// Device Memory Access
+		ReadRegCmd = 0x0080,
+		ReadRegAck = 0x0081,
+		WriteRegCmd = 0x0082,
+		WriteRegAck = 0x0083,
+		ReadMemCmd = 0x0084,
+		ReadMemAck = 0x0085,
+		WriteMemCmd = 0x0086,
+		WriteMemAck = 0x0087,
+		PendingAck = 0x0089,
+
+		// Asynchronous Events
+		EventCmd = 0x00C0,
+		EventAck = 0x00C1,
+		EventDataCmd = 0x00C2,
+		EventDataAck = 0x00C3,
+
+		// Miscellaneous
+		ActionCmd = 0x0100,
+		ActionAck = 0x0101,
+		Unknown = 0xFFFF
+	};
+
+	/// output operator for GvcpCommand
+	std::ostream& operator<<(std::ostream& os, GvcpCommand command);
+
+	/// @brief GVCP response status can be returned in an acknowledge message or a GVSP header.
+	/// See more in the spec "Table 19-1: List of Standard Status Codes"
+	enum class GvcpResponseStatus : uint16_t
```

Comment:
This enum has doxygen documentation for its values but `GvcpCommand` doesn't. I think we should be consistent - either we add doxygen to all values in all enums, or we don't...

Reply:
Some enums in the spec have explanations but some don't. I will add the comments for the enum values as long as the spec has the sentences for the values.